eng/release/DiagnosticsReleaseTool/Common/AzureBlobPublisher.cs

```diff
@@ -0,0 +1,267 @@
+                        await Task.Delay(loopDelay, ct);
+                    }
+                }
+                catch (Exception ex)
+                {
+                    // Azure errors have their own built-in retry logic, so just abort if we got an AzureResponseException
+                    _logger.LogWarning(ex, $"Failed to publish {fileMap.LocalSourcePath}, unexpected error, aborting.");
+                    return null;
+                }
+            } while (retriesLeft > 0 && !completed);
+
+            return result?.OriginalString;
+        }
+
+        private static string GetBlobName(string releaseName, string relativeFilePath)
+        {
+            return $"{releaseName}/{relativeFilePath}";
+        }
+
+        private async Task<BlobContainerClient> GetClient(CancellationToken ct)
+        {
+            if (_client == null)
+            {
+                BlobServiceClient serviceClient = new BlobServiceClient(AccountBlobUri, AccountCredential, BlobOptions);
+                _logger.LogInformation($"Attempting to connect to {serviceClient.Uri} to store blobs.");
+
+                BlobContainerClient newClient;
+                int attemptCt = 0;
+                do
+                {
+                    try
+                    {
+                        newClient = (await serviceClient.CreateBlobContainerAsync(_containerName, PublicAccessType.None, metadata: null, ct)).Value;
+                        _logger.LogInformation($"Created new contianer: {_containerName}.");
+                    }
+                    // 409 is the error code for an existing blob container already existing
+                    catch (RequestFailedException rfe) when (rfe.Status == 409)
+                    {
+                        _logger.LogInformation($"Found existing contianer: {_containerName}.");
+                        // Got an existing container
+                        newClient = serviceClient.GetBlobContainerClient(_containerName);
+                    }
+                    catch (Exception ex)
+                    {
+                        _logger.LogWarning(ex, $"Failed to create or access {_containerName}, retrying with new name.");
+                        continue;
+                    }
+
+                    try
+                    {
+                        // Add the new (or update existing) "download" policy to the container
+                        // This is used to mint the SAS tokens without an expiration policy
+                        // Expiration can be added later by modifying this policy
+                        BlobSignedIdentifier downloadPolicyIdentifier = new BlobSignedIdentifier()
+                        {
+                            Id = AccessPolicyDownloadId,
+                            AccessPolicy = new BlobAccessPolicy()
+                            {
+                                Permissions = "r",
+                                PolicyStartsOn = new DateTimeOffset(DateTime.UtcNow),
```

Comment:
Note that due to clock skew, if you set this policy and immediately attempt to access the resource it may fail. Consider using UtcNow - 1 min or something similar, unless this isn't really a concern here.

Reply:
I added a 15 minute clock skew compensation, this is the standard that Azure recommends.